### PR TITLE
Update tests to replace guava with cactoos #888

### PR DESCRIPTION
**Pull request by Serranya**
In this PR I replaced several occurences of Joiner with JoinedText. See #888

I intentionally did not touch the other idiomatic problems in the tests cases.

I left a puzzle that describes the tasks that still have to be done.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#960](https://github.com/yegor256/takes/issues/960) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#960](https://github.com/yegor256/takes/issues/960) is assigned to @fabriciofx[/z](https://www.0crat.com/u/fabriciofx), here is [why](https://www.0crat.com/footprint/C3FFK3YAY/1ccbfa5d-bfa3-4c6c-b257-dce2de00ac50); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by Serranya** [comment]
@fabriciofx please take a look at this. The task was assigned to you 10 days ago.

**Comment by paulodamaso** [comment]
@0crat assign me

File: src/test/java/org/takes/http/BkBasicTest.java [hunk 1]
```diff
@@ -273,12 +274,13 @@ public void run() {
...
                 socket.getOutputStream().write(
-                    Joiner.on(BkBasicTest.CRLF).join(
+                    new JoinedText(
+                        BkBasicTest.CRLF,
                         BkBasicTest.POST,
                         BkBasicTest.HOST,
                         "",
                         text
-                    ).getBytes()
+                    ).asString().getBytes()
```
Comment (paulodamaso) [inline src/test/java/org/takes/http/BkBasicTest.java]: @Serranya Let's use `cactoos` `BytesOf` here WDYT?

File: src/test/java/org/takes/http/BkBasicTest.java [hunk 2]
```diff
@@ -327,13 +329,14 @@ public void run() {
...
-                    Joiner.on(BkBasicTest.CRLF).join(
+                    new JoinedText(
+                        BkBasicTest.CRLF,
                         BkBasicTest.POST,
                         BkBasicTest.HOST,
                         "Connection: Close",
                         "",
                         greetings
-                    ).getBytes()
+                    ).asString().getBytes()
```
Comment (paulodamaso) [inline src/test/java/org/takes/http/BkBasicTest.java]: @Serranya Here too, please use `BytesOf`

File: src/test/java/org/takes/http/BkBasicTest.java [hunk 3]
```diff
@@ -359,13 +362,14 @@ public void run() {
...
-                Joiner.on(BkBasicTest.CRLF).join(
+                new JoinedText(
+                    BkBasicTest.CRLF,
                     "GET / HTTP/1.1",
                     BkBasicTest.HOST,
                     "Content-Length: 2",
                     "",
                     "hi"
-                ).getBytes()
+                ).asString().getBytes()
```
Comment (paulodamaso) [inline src/test/java/org/takes/http/BkBasicTest.java]: @sebing `BytesOf` here too

**Comment by 0crat** [comment]
> @0crat assign me ([here](https://github.com/yegor256/takes/issues/960#issuecomment-465774545))

@paulodamaso @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) not enough funds available in the project, can't set budget of job [#960](https://github.com/yegor256/takes/issues/960), see [§21](http://www.zerocracy.com/policy.html#21); @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) will get no money on completion; in order to fix that, add funds to the project and assign the job again

**Comment by 0crat** [comment]
Manual assignment of issues is discouraged, see [§19](http://www.zerocracy.com/policy.html#19): -5 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by Serranya** [comment]
@paulodamaso I addressed your comments

**Comment by paulodamaso** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job [#960](https://github.com/yegor256/takes/issues/960) is not in the agenda of @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), can't retrieve data and time of add

**Comment by 0crat** [comment]
Job was finished in 19 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
The job [#960](https://github.com/yegor256/takes/issues/960) is now out of scope

**Comment by 0crat** [comment]
Order was finished: +20 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by ypshenychka** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/yegor256/takes/issues/960#issuecomment-466180082))

@ypshenychka This is what I know about this job in [C3FFK3YAY](https://www.0crat.com/p/C3FFK3YAY), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#960](https://github.com/yegor256/takes/issues/960) is not in scope
  * Job [footprint](https://www.0crat.com/footprint/C3FFK3YAY?q=%7Bjob%3A%27gh%3Ayegor256%2Ftakes%23960%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Pull request merged**